### ISSUE TITLE
Gallery app does not work with Scality/Ceph

### ISSUE THREAD

**Issue by davitol** [owncloud/files_primary_s3#72]
# Steps to reproduce

1. User uploads some pics to ownCloud
2. Go to Gallery App 

## Screenshot
![screen shot 2018-05-31 at 14 18 21](https://user-images.githubusercontent.com/10596560/40781981-28834492-64de-11e8-8f4c-4013d60d280e.png)

## Tested with
OwnCloud based on PR https://github.com/owncloud/core/pull/31050 and files_primary_s3

Behavior is not observed on ownCloud 10.0.8 docker containers:
![screen shot 2018-05-31 at 14 22 37](https://user-images.githubusercontent.com/10596560/40781964-1e320104-64de-11e8-9b4f-c26a0283a99b.png)

**Comment by patrickjahns** [comment]
checking logs in the instance, i saw this related to the error:
```
{"reqId":"nC9glRP2AwS4pt18VcCM","level":3,"time":"2018-05-31T12:11:59+00:00","remoteAddr":"172.18.0.4","user":"admin","app":"index","method":"GET","url":"\/apps\/gallery\/thumbnails?ids=31%3B32%3B35%3B40%3B44%3B46&scale=1&square=0&requesttoken=AVM3Fis%2BBy8VJQgpXwBXUSwDMTVyFBF%2BHjcvUwMFHSc%3D%3Ak2SEfpfEQFPX2F5aopXyFwEQxun8g4hltzUDB2RXKlw%3D","message":"Exception: {\"Exception\":\"TypeError\",\"Message\":\"Argument 3 passed to OC\\\\Preview::__construct() must implement interface OCP\\\\Files\\\\Node or be null, string given, called in \\\/mnt\\\/data\\\/apps\\\/gallery\\\/preview\\\/preview.php on line 100\",\"Code\":0,\"Trace\":\"#0 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/preview\\\/preview.php(100): OC\\\\Preview->__construct('admin', 'files', 'Screen Shot 201...')\\n#1 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/service\\\/previewservice.php(107): OCA\\\\Gallery\\\\Preview\\\\Preview->setupView('admin', Object(OC\\\\Files\\\\Node\\\\File), 'Screen Shot 201...')\\n#2 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/controller\\\/preview.php(167): OCA\\\\Gallery\\\\Service\\\\PreviewService->createPreview(Object(OC\\\\Files\\\\Node\\\\File), 400, 200, true, true)\\n#3 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/controller\\\/preview.php(117): OCA\\\\Gallery\\\\Controller\\\\PreviewController->getPreviewData(Object(OC\\\\Files\\\\Node\\\\File), false, 400, 200, true, true)\\n#4 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/controller\\\/preview.php(86): OCA\\\\Gallery\\\\Controller\\\\PreviewController->getData(31, 400, 200, true, false, true)\\n#5 \\\/mnt\\\/data\\\/apps\\\/gallery\\\/controller\\\/previewcontroller.php(103): OCA\\\\Gallery\\\\Controller\\\\PreviewController->getThumbnail(31, false, '1')\\n#6 [internal function]: OCA\\\\Gallery\\\\Controller\\\\PreviewController->getThumbnails('31;32;35;40;44;...', false, '1')\\n#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(159): call_user_func_array(Array, Array)\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(89): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\Gallery\\\\Controller\\\\PreviewController), 'getThumbnails')\\n#9 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(103): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\Gallery\\\\Controller\\\\PreviewController), 'getThumbnails')\\n#10 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('PreviewControll...', 'getThumbnails', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#11 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(342): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(912): OC\\\\Route\\\\Router->match('\\\/apps\\\/gallery\\\/t...')\\n#14 \\\/var\\\/www\\\/owncloud\\\/index.php(55): OC::handleRequest()\\n#15 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Preview.php\",\"Line\":112}"}
```

**Comment by DeepDiver1975** [comment]
fix is in here - https://github.com/owncloud/gallery/pull/738/files#diff-9ca667cac53731526d8c0dd25c5d4ac0R101

**Comment by patrickjahns** [comment]
That means however - we need a clean cut for `files_primary_s3` and gallery to work together right? 
Or we need to adjust that PR again to have it use the ownCloud patch version ( 👎 )

**Comment by DeepDiver1975** [comment]
all we need to do is to release the gallery app before we push 10.0.9 which is planned to bring the relevant core changes.

**Comment by DeepDiver1975** [comment]
@davitol @patrickjahns we can close this from my pov

**Comment by davitol** [comment]
> fix is in here - https://github.com/owncloud/gallery/pull/738/files#diff-9ca667cac53731526d8c0dd25c5d4ac0R101

Tested with Gallery master branch and works

**Comment by davitol** [comment]
Fixed in S3 Object Storage 1.0.0
